### PR TITLE
Add mechanism to mark an operator as deprecated in favor of another one

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

**Comment by JanuszL** [comment]
Can you provide a screen shot how it looks like in the docs?

**Comment by jantonguirao** [comment]
> What about appending the deprecation note to the docstring?

@Kh4L isn't it what I did with my changes in `ops.py`?

**Comment by jantonguirao** [comment]
Screenshot from 2019-06-24 08-12-07

@JanuszL ^^^

**Comment by jantonguirao** [comment]
!build

**Comment by JanuszL** [comment]
> Screenshot from 2019-06-24 08-12-07
> 
> @JanuszL ^^^

I meant sphinx docs:
image

I think we should highlight it something.

**Comment by JanuszL** [comment]
How about http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated ?

**Comment by jantonguirao** [comment]
> How about http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated ?

used `.. warning:` as agreed

**Comment by jantonguirao** [comment]
!build

File: dali/pipeline/operators/op_schema.h [hunk 1]
```diff
@@ -187,6 +188,15 @@ class DLL_PUBLIC OpSchema {
     return *this;
   }
 
+  /**
+   * @brief Notes that this operator is deprecated and optionally specifies the operator to be used instead
+   */
+  DLL_PUBLIC inline OpSchema& MakeDeprecatedInFavorOf(const std::string &deprecated_in_favor_of) {
```
Comment (mzient) [inline dali/pipeline/operators/op_schema.h]:
```suggestion
  DLL_PUBLIC inline OpSchema& Deprecate(const std::string &n_favor_of = "") {
```

Comment (jantonguirao) [reply dali/pipeline/operators/op_schema.h]: done

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
Screenshot from 2019-07-01 09-35-08

**Comment by jantonguirao** [comment]
!build

**Pull request merged**